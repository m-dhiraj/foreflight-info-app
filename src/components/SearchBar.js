import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { fetchAirportInfo } from '../redux/airportSlice';
import '../styles/SearchBar.css'

const SearchBar = () => {
    // Hook to store and update the search
    const [airportCode, setAirportCode] = useState('');
    const dispatch = useDispatch();

    const handleSearch = () => {
        // use airport code to dispatch the fetchAirportInfo action 
        if (airportCode.trim()) {
            dispatch(fetchAirportInfo(airportCode.trim()));
        }
    };
    // render the actul search bar
    return (
        <div className="search-bar-container">
            <input
            type="text"
            value={airportCode}
            onChange={(e) => setAirportCode(e.target.value.toUpperCase())}
            placeholder="Enter ICAO airport code"
            />
            <button onClick={handleSearch}>Search</button>
        </div>
    );
};

export default SearchBar;
