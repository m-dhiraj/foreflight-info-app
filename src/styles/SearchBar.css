.search-bar-container {
    text-align: center;
    margin: 20px;
    padding: 20px;
}

.search-bar-container input {
    padding: 10px 15px;
    font-size: 16px;
    border-radius: 4px;
    border: 1px solid #ccc;
    width: 300px;
    display: inline-block;
    margin-right: 10px;
}

.search-bar-container button {
    padding: 10px 20px;
    font-size: 16px;
    background-color: blue;
    color: white;
    border: none;
    border-radius: 4px;
    cursor: pointer;
    transition: background-color 0.3s ease;
}

.search-bar-container button:hover {
    background-color: darkblue;
}
