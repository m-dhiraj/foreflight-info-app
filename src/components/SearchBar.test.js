import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import airportReducer from '../redux/airportSlice';
import SearchBar from './SearchBar';

const store = configureStore({
  reducer: {
    airports: airportReducer
  }
});

describe('SearchBar Component', () => {
  it('renders correctly and matches the snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <SearchBar />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
