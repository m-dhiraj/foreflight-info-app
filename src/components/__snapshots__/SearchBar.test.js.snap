// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component renders correctly and matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="search-bar-container"
  >
    <input
      placeholder="Enter ICAO airport code"
      type="text"
      value=""
    />
    <button>
      Search
    </button>
  </div>
</DocumentFragment>
`;
